esphome/components/cm1106/cm1106.cpp

```diff
@@ -0,0 +1,111 @@
+#include "cm1106.h"
+#include "esphome/core/log.h"
+
+#include <cinttypes>
+
+namespace esphome {
+namespace cm1106 {
+
+static const char *const TAG = "cm1106";
+
+uint8_t cm1106_checksum(const uint8_t *response, size_t len) {
+  uint8_t crc = 0;
+  for (int i = 0; i < len - 1; i++) {
+    crc -= response[i];
+  }
+  return crc;
+}
+
+void CM1106Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up CM1106...");
+  uint8_t response[8] = {0};
+  if (!this->cm1106_write_command_(this->c_m1106_cmd_get_co2_, sizeof(this->c_m1106_cmd_get_co2_), response,
+                                   sizeof(response))) {
+    ESP_LOGW(TAG, "Reading data from CM1106 failed!");
```

Comment:
```suggestion
    ESP_LOGE(TAG, "Reading data from CM1106 failed!");
```
This is actually error, it happened in setup.